contrib/linux/bash-completion/dosbox

```diff
@@ -0,0 +1,183 @@
+# vim: st=2:sw=2:et:ai:ft=bash
+### EDITING NOTES:
+# As this is a bash function, when adding variables or arrays it is important
+# to initialise with local. This prevents the var./array polluting the users
+# env. Please check output of:
+#    declare -x ; declare -a
+# after testing your new completion script
+#
+# Notable exception is COMPREPLY, this needs to persist for communication with
+# bash_completion
+####
+_dosbox()
+{
+  COMPREPLY=()
+  local cur prev words cword
+  _init_completion -s || return
+  local IFS=$'|\t\n'
+  local words_filter="${words[*]:1:$(( cword - 1 ))}"
+
+  ### Main options, note the \ line cont.
+  # using nameref
+  # shellcheck disable=SC2034
+  local -a \
+       _main_opts=(
+                    -h --help -fullscreen -startmapper -noautoexec -securemode
+                    -scaler -forcescaler -lang -machine -socket -exit
+                  )\
+     _repeat_opts=(
+                    -c -conf
+                  )\
+  _exclusive_opts=(
+                    --version --printconf --editconf -resetconf -eraseconf
+                    -opencaptures -erasemapper -resetmapper
+                  )\
+  ####
+
+  # setup nameref to point to array
+  [[ ${prev:0:1} == - ]] && local -n _sub_opts=${prev//+(-)/_}
+
+  ### Arrays holding sub. options
+  # The name should be the trigger option with leading "-"s replaced with "_"
+  # for example, --machine => _machine
+  # latter parts of the script will reference these arrays via a "nameref"
+  # i.e. The last option dictates the available sub. options
+
+  #
+  # shellcheck disable=SC2034
+  local -a \
+         _machine=(
+                    hercules cga tandy pcjr ega vgaonly svga_s3 svga_et3000
+                    svga_et4000 svga_paradise vesa_nolfb vesa_oldvbe
+                  )\
+          _scaler=(
+                    none normal2x normal3x advmame2x advmame3x advinterp2x
+                    advinterp3x hq2x hq3x 2xsai super2xsai supereagle tv2x
+                    tv3x rgb2x rgb3x scan2x scan3x
+                  )\
+    _opencaptures=(
+                    "$BROWSER" firefox mpv pcmanfm pcmanfm-qt
+                  )\
+  # A note on -opencaptures, it launches the provided prog. and points it to
+  # the dir. containing captured media (screenshots, video, audio)
+  # TODO add more media browser programs
+
+  [[ $prev =~ ^-{1,2}opencaptures$ ]] &&
+    readarray -t -O 0 _opencaptures < <(
+      # do we have a custom wrapper script available?
+      [[ -x "$XDG_CONFIG_HOME/dosbox/opencaptures.sh" ]] &&
+        printf "%s\n" "$XDG_CONFIG_HOME/dosbox/opencaptures.sh"
+      # TODO provide example script
+      # may not be worth it..
+
+      # filter out unavailable media browsers
+      type -P "${_opencaptures[@]}"
+      # TODO document recommended "media browsers"
+    )
+  ####
+
+  ### Offer ability for the user to override the default .conf dir.
+  # shellcheck disable=SC2034
+  local _conf="${DOSBOX_STAGING_CONF_DIR:-$XDG_CONFIG_HOME/dosbox/}"
```

Comment:
No, please don't. XDG_CONFIG_HOME is already a mechanism to override the default conf dir - we DO NOT want to invent a new one.

Reply:
Yeah, I can drop that, I never intended it to be documented, it was just a way for me to dump a bunch of conf files someplace without polluting my env.